src/core/providers/proxy.ts

```diff
@@ -3,7 +3,10 @@ import { ChainId } from '../types/chains';
 export const proxyRpcEndpoint = (endpoint: string, chainId: ChainId) => {
   if (
     endpoint !== 'http://127.0.0.1:8545' &&
-    endpoint !== 'http://localhost:8545'
+    endpoint !== 'http://localhost:8545' &&
+    !endpoint.includes('http://10.') &&
+    !endpoint.includes('http://192.') &&
+    !endpoint.includes('http://172.')
```

Comment:
IANA private ips range is 192.168, so 192.16**9**.x.x can be public
and 172. ranges from 172.16.x.x to 172.31.x.x

maybe something like this
```suggestion
    !endpoint.includes('http://192.168') &&
    !endpoint.match(/http:\/\/172.(1[6-9]|2[0-9]|3[0-1])./)
```

Reply:
Right. This seems okay.